### PR TITLE
docs: Document Potential Dual-Stack Upgrade Issues for 1.15

### PR DESCRIPTION
**Pull request by nathanjsweet**
Depends on https://github.com/cilium/cilium/pull/22625

### CODE REVIEW

File: Documentation/operations/upgrade.rst
Comment (zacharysarah) [inline Documentation/operations/upgrade.rst]: No Latin abbreviations
```suggestion
* A bug that prevented dual-stack clusters (clusters that use both IPv4
```

File: Documentation/operations/upgrade.rst
Comment (zacharysarah) [inline Documentation/operations/upgrade.rst]: Edits for clarity, use present tense
```suggestion
  fixed. If operators of dual-stack clusters need to
  roll back from 1.14 (for example, because of a failed upgrade), existing connections to
  endpoints external to the cluster may break, though new connections will
  work correctly. Upgrading to 1.14 does not cause this problem.
```

**Review by zacharysarah — CHANGES_REQUESTED** [review]
@nathanjsweet Thanks for the update. Some small edits required, otherwise LGTM.

File: Documentation/operations/upgrade.rst
Comment (zacharysarah) [inline Documentation/operations/upgrade.rst]: I know it looks inconsistent with my "use present tense" advice, but in this case past tense is correct.
```suggestion
* A bug that prevented dual-stack clusters (clusters that use both IPv4
  and IPv6 addresses) from referencing the total address space of each IP
  family individually ("0.0.0.0/0" and "::/0") in network policies has been
  fixed. If operators of dual-stack clusters need to roll back from 1.14 (for
  example, because of a failed upgrade) existing connections to
  endpoints external to the cluster may break, though any new connections will
  work correctly. Upgrading to 1.14 does not cause this problem.
```

**Review by zacharysarah — DISMISSED** [review]
@nathanjsweet 👋🏻 Thanks for the fixes! Two small nits, otherwise LGTM.

I'm approving this PR for docs with the understanding that changes must be made prior to merge. 🙇🏻

File: Documentation/operations/upgrade.rst
Comment (joestringer) [inline Documentation/operations/upgrade.rst]: Given that this missed v1.14.0, this should probably be in a new section above.

File: Documentation/operations/upgrade.rst
Comment (joestringer) [inline Documentation/operations/upgrade.rst]: Reading afresh, I'm struggling a bit with understanding this from a user perspective. 

Key things to know are:
- Am I affected? Which settings do I need to have enabled?
- Under which conditions will I hit the issue?
- How do I know if the problem I'm facing is expected or not?
- How do I know when the problem is resolved?
- What should I do if the problem is not resolved?

I think that a lot of the points above are covered, but we could potentially make this a bit clearer for users. For example:

```suggestion
* If you configured Cilium with IPv4 and IPv6 support enabled, and you have a network policy with a ``ToCIDR`` or ``ToCIDRSet`` rule matching a full IP range such as ``0.0.0.0/0`` or ``::/0``, then you may experience connection breakage when switching Cilium versions. When this problem occurs, existing connections allowed by the network policy may be denied until the application reconnects. New connections are not impacted. Upgrading from Cilium 1.14.x or earlier to 1.15.y or later does not trigger this problem. Downgrading from Cilium 1.15.y or later to Cilium 1.14.x or earlier may trigger this problem. 
```

I think this is more direct to the point to inform the user in the first few statements exactly which conditions they need to know about their cluster to see if they will hit the issue, and if not they can ignore the note. If they think they may be affected, they will read on to understand the limits of the impact and the expected behaviour.

If you think this wording is clearer or there is a variation on the above that is clearer, feel free to adopt (some of) the above text. I happen to think that this is a bit clearer, but I may be biased by the fact I just wrote the sentences myself, so I'll leave it to your discretion on whether to make changes to the release note.

(That said, I must admit I don't recall the details of why this only impacts Cilium pods upon downgrade, since Cilium should recalculate the correct state after downgrade and the identity for connection peers shouldn't be held in the connection tracker.. but I think that the above accurately reflects the statements you were describing previously in the commit)

**Review by joestringer — APPROVED** [review]
LGTM

**Pull request merged**
